src/main/java/seedu/address/logic/CommandHistory.java

```diff
@@ -58,6 +71,24 @@ public String getNext() {
         return ""; // return blank line when pressing down after most recent command is reached
     }
 
+    /**
+     * Removes latest command from commandInputHistory.
+     */
+    public void remove() {
+        commandInputHistory.remove(commandInputHistory.size() - 1);
+    }
+
+    /**
+     * Returns an unmodifiable view of {@code commandInputHistory}.
+     */
+    public ObservableList<Command> getCommandHistory() {
+        return unmodifiableCommandInputHistory;
+    }
+
+    public int getSize() {
```

Comment:
I dont't think this method is necessary since the user can just call the getter method above to get the list and call the `getSize` method directly.